src/PAModel/EditorState/ControlState.cs

```diff
@@ -14,12 +11,11 @@ internal class ControlState
     {
         public string Name { get; set; }
 
-        [JsonIgnore]
```

Comment:
Are we writing this for every control in the editor state file? Is there a way we could only write it once?

Reply:
I believe that yes, it is being written as a property for every control. Let me dig on this a bit more to see what we can do here.

Reply:
I've reworked this so that we're not storing it on each control but instead denoting the top parent control only. To avoid adding additional time complexity when reading in the editor state during deserialization, I've also used a well-known name for the dictionary key of the top parent control. This key was otherwise not being used, and lets us immediately find the top parent and use its name.